release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,321 @@
+            }
+        });
+        System.out.println(builder.toString());
+    }
+
+    String getFormattedReleaseNotes(Category category, List<ReleaseNoteText> releaseNotes) {
+        StringBuilder builder = new StringBuilder();
+        releaseNotes.forEach(rnText -> {
+            builder.append(String.format("- %s (", rnText.text));
+            rnText.prNumbers.forEach(prNum -> {builder.append(String.format("#%s ", prNum));});
+            builder.deleteCharAt(builder.length() - 1); // delete the last space character
+            builder.append(")\n");
+        });
+        return builder.toString();
+    }
+
+    void checkReleaseNoteCategory(ReleaseNoteText rnText) {
+        if (rnText.category == null) {
+            rnText.category = Category.MISCELLANEOUS;
+        }
+    }
+
+    boolean isContainedInSameAsItems(ReleaseNoteText rnText) {
+        return sameAsItems.values().stream().anyMatch(items -> {
+                return items.contains(rnText);
+            });
+    }
+
+    enum Category {
+        ENHANCEMENT("Enhancements"),
+        IMPROVEMENT("Improvements"),
+        BUGFIX("Bug fixes"),
+        DOCUMENTATION("Documentation"),
+        MISCELLANEOUS("Miscellaneous");
+
+        private final String displayName;
+
+        Category(String displayName) {
+            this.displayName = displayName;
+        }
+
+        public String getDisplayName() {
+            return this.displayName;
+        }
+
+        public static Category getByName(String name) {
+            return Arrays.stream(Category.values())
+                    .filter(v -> v.name().toLowerCase().equals(name))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException("Invalid name: " + name));
+        }
+
+        public static Category getByDisplayName(String displayName) {
+            return Arrays.stream(Category.values())
+                    .filter(v -> v.displayName.equals(displayName))
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException("Invalid displayName: " + displayName));
+        }
+    }
+
+    class ReleaseNoteText {
+        public Category category;
+        public String text;
+        public List<String> prNumbers = new ArrayList<>();
+    }
+
+    class GitHubContext {
+
+        private final String MERGED_STATE = "merged";
+
+        private String owner = "scalar-labs";
+        private String projectTitlePrefix = "ScalarDB";
+        private String version;
+        private String repository;
+
+        public GitHubContext(String owner, String projectTitString, String version, String repository) {
+            this.owner = owner;
+            this.projectTitlePrefix = projectTitString;
+            this.version = version;
+            this.repository = repository;
+        }
+        
+        public String getProjectId() throws Exception {
+            BufferedReader br = runSubProcessAndGetOutputAsReader(
```

Comment:
I'm not sure if it works, but using Java SDK for GitHub such as https://github.com/hub4j/github-api might be a bit more robust?

Reply:
Maybe so. But I wrote the script this way because I wanted it to be executable as [Single-File Source-Code Program](https://www.infoq.com/articles/single-file-execution-java11/) so we could easily run it with only java command.